### PR TITLE
Red 40846 plugin crashes on yosemite

### PR DESCRIPTION
**Pull request by bharath-nadampalli**

### CODE REVIEW

**Comment by bharath-nadampalli** [comment]
Please create the NativityFinder dll both client and finder plugin and checkin

File: csharp/src/Liferay.Nativity/Control/Unix/UnixNativityControlBaseImpl.cs [hunk 1]
```diff
@@ -68,62 +74,103 @@ public class UnixNativityControlBaseImpl : NativityControl
...
+					logger.DebugFormat(
+						"Successfully connected to service socket: {0}",
+						UnixNativityControlBaseImpl.CALLBACK_SOCKET_PORT);
+
+					return true;
+				}
+				catch (Exception e)  // IOException???
+				{
+					logger.Info("Connect Exception");
+					logger.Error(e);
```
Comment (andysyncplicity) [inline csharp/src/Liferay.Nativity/Control/Unix/UnixNativityControlBaseImpl.cs]: You don't need two calls to the logger. Just do something like:

logger.Error("Exception connecting", e);

File: csharp/src/Liferay.Nativity/Control/Unix/UnixNativityControlBaseImpl.cs [hunk 2]
```diff
@@ -68,62 +74,103 @@ public class UnixNativityControlBaseImpl : NativityControl
...
+
+		public override void CheckSocketConnection()
+		{
+			try
+			{
+
+				if((this.commandSocket.Available == 0 && commandSocket.Client.Poll((int)SOCKETCONNECTED_POLL_TIME.TotalMilliseconds, SelectMode.SelectRead)) ||
+					this.callbackSocket.Available == 0 && callbackSocket.Client.Poll((int)SOCKETCONNECTED_POLL_TIME.TotalMilliseconds, SelectMode.SelectRead))
+				{
+					logger.Info("CheckConnection failed restarting connection.");
```
Comment (andysyncplicity) [inline csharp/src/Liferay.Nativity/Control/Unix/UnixNativityControlBaseImpl.cs]: Please log this to logger.Warn

File: csharp/src/Liferay.Nativity/Control/Unix/UnixNativityControlBaseImpl.cs [hunk 3]
```diff
@@ -68,62 +74,103 @@ public class UnixNativityControlBaseImpl : NativityControl
...
+					var checkMessageString =  JsonConvert.SerializeObject(message);
+					this.callbackOutputStream.WriteLine(checkMessageString);
+				}
 			}
-			catch (Exception e)  // IOException???
+			catch(Exception ex)
 			{
-				logger.Error(e);
-				this.connected = false;
+				logger.Info("CheckConnection failed restarting connection : ", ex);
```
Comment (andysyncplicity) [inline csharp/src/Liferay.Nativity/Control/Unix/UnixNativityControlBaseImpl.cs]: Please log to logger.Error

File: csharp/src/Liferay.Nativity/Control/Unix/UnixNativityControlBaseImpl.cs [hunk 4]
```diff
@@ -68,62 +74,103 @@ public class UnixNativityControlBaseImpl : NativityControl
...
+						"Successfully connected to service socket: {0}",
+						UnixNativityControlBaseImpl.CALLBACK_SOCKET_PORT);
+
+					return true;
+				}
+				catch (Exception e)  // IOException???
+				{
+					logger.Info("Connect Exception");
+					logger.Error(e);
+					this.Dispose();
```
Comment (andysyncplicity) [inline csharp/src/Liferay.Nativity/Control/Unix/UnixNativityControlBaseImpl.cs]: It's a little odd to call Dispose and then continue using an object. Why don't you just call this.Disconnect()? If we ever refactor the base class, we could end up getting a lot of ObjectDisposedExceptions.

File: mac/LiferayNativityFinder/GCDAsyncSocket.m [hunk 5]
```diff
@@ -2504,8 +2504,9 @@ - (void)doConnectTimeout
 
 - (void)closeWithError:(NSError *)error
 {
+	NSLog(@"closeWithError");
```
Comment (andysyncplicity) [inline mac/LiferayNativityFinder/GCDAsyncSocket.m]: GDCAsyncSocket is an open-source library that the LiferayNativity project used.

Do you know if they fixed this?

(I remember when I tried using a newer GCDAsyncSocket, I had issues.)

File: mac/LiferayNativityFinder/RequestManager.m [hunk 6]
```diff
@@ -696,10 +744,11 @@ - (void)socketDidDisconnect:(GCDAsyncSocket*)socket withError:(NSError*)err
 {
 	// This callback can happen on either queue, yet each queue has private data
 	// In order to ensure thread-safe reads from each collection, perform the actual disconnect logic on the appropriate queue
-	
+	NSLog(@"Entering socketDidDisconnect");
```
Comment (andysyncplicity) [inline mac/LiferayNativityFinder/RequestManager.m]: Should this only be logged when debug logging is enabled?

**Pull request merged**